File: seeker/snippet/mergesort.py
#date: 2022-01-11T17:21:17Z
#url: https://api.github.com/gists/b9570fe21b0315c6288ff7432f5a4b73
#owner: https://api.github.com/users/dilettagoglia

# MERGESORT ALGORITHM

import numpy

x=(numpy.random.randint(0, 33))
y=(numpy.random.randint(0, 33))
A=numpy.random.randint(x, size=y)
temp = numpy.zeros(len(A), dtype=int)

def merge(V, l, c, r):
  i = k = l
  j = c+1
  while i<=c and j<=r:
    if V[i] <= V[j]:
      temp[k] = V[i]
      i = i+1
    else:
      temp[k] = V[j]
      j= j+1
    k = k+1
  while i<=c:
    temp[k] = V[i]
    i = i+1
    k = k+1
  while j<=r:
    temp[k] = V[j]
    j = j+1
    k = k+1
  for k in range(l, r+1):
    V[k] = temp[k]

def mergesort(V, l, r):
  if l>=r:
    return
  m = (l+r)//2
  mergesort(V, l, m)
  mergesort(V, m+1, r)
  merge(V, l, m, r)
  return V

File: seeker/snippet/mergeSort.java
//date: 2023-05-25T16:42:22Z
//url: https://api.github.com/gists/1a3af7dab6841737fb6067d4a9f64908
//owner: https://api.github.com/users/ashutoshdhande

import java.util.Arrays;

/**
 * MergeSort
 */
class MergeSort {
     public static void main(String[] args) {
          int[] arr = { 2, 4, 1, 6, 8, 5, 3, 7 };
          mergeSort(arr, arr.length);
          System.out.println(Arrays.toString(arr));
     }

     static void mergeSort(int[] arr, int n) {
          // int n = arr.length;
          if (n < 2)
               return;

          int mid = n / 2;
          int[] left = new int[mid];
          int[] right = new int[n - mid];

          for (int i = 0; i < mid; i++) {
               left[i] = arr[i];
          }
          for (int i = mid; i < n; i++) {
               right[i - mid] = arr[i];
          }

          mergeSort(left, mid);
          mergeSort(right, n - mid);
          merge(left, right, arr);
     }

     static void merge(int[] left, int[] right, int[] arr) {
          // int n = arr.length;
          int nl = left.length;
          int nr = right.length;
          int i = 0, j = 0, k = 0;
          while (i < nl && j < nr) {
               if (left[i] < right[j]) {
                    arr[k++] = left[i++];

               } else {
                    arr[k++] = right[j++];

               }
          }

          while (i < nl) {
               arr[k++] = left[i++];

          }

          while (j < nr) {
               arr[k++] = right[j++];

          }

     }

}